### ISSUE TITLE
Pkg.add fails for DifferentialEquations: error building SpecialFunctions and Sundials

### ISSUE THREAD

**Issue by rgalarcia** [JuliaLang/Pkg.jl#1243]
I am trying to add the DifferentialEquations package to a fresh installation of Julia 1.1.1 on a Debian GNU/Linux 9.9 (stretch) machine, by using the following commands:

>using Pkg
>Pkg.add("DifferentialEquations")

The latter command takes ages to run (about 30 minutes trapped in "Resolving package versions…") and then it fails to build SpecialFunctions and Sundials, which I understand that are package dependencies. The output is the following:

```
julia> Pkg.add("DifferentialEquations")
   Cloning default registries into `~/.julia`
   Cloning registry from "https://github.com/JuliaRegistries/General.git"
     Added registry `General` to `~/.julia/registries/General`
 Resolving package versions...
   Cloning [0c46a032-eb83-5123-abaf-570d42b7fbaa] DifferentialEquations from https://github.com/JuliaDiffEq/DifferentialEquations.jl.git
[ Info: Installed DifferentialEquations ─────── v6.4.0
   Cloning [49dc2e85-a5d0-5ad3-a950-438e2897f1b9] Calculus from https://github.com/JuliaMath/Calculus.jl.git
[ Info: Installed Calculus ──────────────────── v0.4.1
   Cloning [b4f34e82-e78d-54a5-968a-f98e89d6e8f7] Distances from https://github.com/JuliaStats/Distances.jl.git
[ Info: Installed Distances ─────────────────── v0.8.0
   Cloning [069b7b12-0de2-55c6-9aab-29f3d0a68a2e] FunctionWrappers from https://github.com/yuyichao/FunctionWrappers.jl.git
[ Info: Installed FunctionWrappers ──────────── v1.0.0
   Cloning [b99e7846-7c00-51b0-8f62-c81ae34c0232] BinaryProvider from https://github.com/JuliaPackaging/BinaryProvider.jl.git
[ Info: Installed BinaryProvider ────────────── v0.5.5
   Cloning [77ba4419-2d1f-58cd-9bb1-8ffee604a2e3] NaNMath from https://github.com/mlubin/NaNMath.jl.git
[ Info: Installed NaNMath ───────────────────── v0.3.2
   Cloning [aae01518-5342-5314-be14-df237901396f] BandedMatrices from https://github.com/JuliaMatrices/BandedMatrices.jl.git
[ Info: Installed BandedMatrices ────────────── v0.9.2
   Cloning [34da2185-b29b-5c13-b0c7-acf172513d20] Compat from https://github.com/JuliaLang/Compat.jl.git
[ Info: Installed Compat ────────────────────── v2.1.0
   Cloning [5078a376-72f3-5289-bfd5-ec5146d43c02] LazyArrays from https://github.com/JuliaArrays/LazyArrays.jl.git
[ Info: Installed LazyArrays ────────────────── v0.9.0
   Cloning [864edb3b-99cc-5e75-8d2d-829cb0a9cfe8] DataStructures from https://github.com/JuliaCollections/DataStructures.jl.git
[ Info: Installed DataStructures ────────────── v0.15.0
   Cloning [01680d73-4ee2-5a08-a1aa-533608c188bb] GenericSVD from https://github.com/JuliaLinearAlgebra/GenericSVD.jl.git
[ Info: Installed GenericSVD ────────────────── v0.2.1
   Cloning [1a297f60-69ca-5386-bcde-b61e274b549b] FillArrays from https://github.com/JuliaArrays/FillArrays.jl.git
[ Info: Installed FillArrays ────────────────── v0.6.3
   Cloning [3cdcf5f2-1ef4-517c-9805-6587b60abb01] RecipesBase from https://github.com/JuliaPlots/RecipesBase.jl.git
[ Info: Installed RecipesBase ───────────────── v0.7.0
   Cloning [42fd0dbc-a981-5370-80f2-aaf504508153] IterativeSolvers from https://github.com/JuliaMath/IterativeSolvers.jl.git
[ Info: Installed IterativeSolvers ──────────── v0.8.1
   Cloning [f9640e96-87f6-5992-9c3b-0743c6a49ffa] MultiScaleArrays from https://github.com/JuliaDiffEq/MultiScaleArrays.jl.git
[ Info: Installed MultiScaleArrays ──────────── v1.4.0
   Cloning [1dea7af3-3e70-54e6-95c3-0bf5283fa5ed] OrdinaryDiffEq from https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl.git
[ Info: Installed OrdinaryDiffEq ────────────── v5.8.1
   Cloning [b552c78f-8df3-52c6-915a-8e097449b14b] DiffRules from https://github.com/JuliaDiff/DiffRules.jl.git
[ Info: Installed DiffRules ─────────────────── v0.0.10
   Cloning [90137ffa-7385-5640-81b9-e52037218182] StaticArrays from https://github.com/JuliaArrays/StaticArrays.jl.git
[ Info: Installed StaticArrays ──────────────── v0.11.0
   Cloning [f2b01f46-fcfa-551c-844a-d8ac1e96c665] Roots from https://github.com/JuliaMath/Roots.jl.git
[ Info: Installed Roots ─────────────────────── v0.8.1
   Cloning [459566f4-90b8-5000-8ac3-15dfb0a30def] DiffEqCallbacks from https://github.com/JuliaDiffEq/DiffEqCallbacks.jl.git
[ Info: Installed DiffEqCallbacks ───────────── v2.5.2
   Cloning [ffbed154-4ef7-542d-bbb7-c09d3a79fcae] DocStringExtensions from https://github.com/JuliaDocs/DocStringExtensions.jl.git
[ Info: Installed DocStringExtensions ───────── v0.8.0
   Cloning [bac558e1-5e72-5ebc-8fee-abe8a469f55d] OrderedCollections from https://github.com/JuliaCollections/OrderedCollections.jl.git
[ Info: Installed OrderedCollections ────────── v1.1.0
   Cloning [764a87c0-6b3e-53db-9096-fe964310641d] BoundaryValueDiffEq from https://github.com/JuliaDiffEq/BoundaryValueDiffEq.jl.git
[ Info: Installed BoundaryValueDiffEq ───────── v2.2.3
   Cloning [d4d017d3-3776-5f7e-afef-a10c40355c18] ExponentialUtilities from https://github.com/JuliaDiffEq/ExponentialUtilities.jl.git
[ Info: Installed ExponentialUtilities ──────── v1.5.1
   Cloning [ae5879a3-cd67-5da8-be7f-38c6eb64a37b] ResettableStacks from https://github.com/JuliaDiffEq/ResettableStacks.jl.git
[ Info: Installed ResettableStacks ──────────── v0.6.0
   Cloning [78ddff82-25fc-5f2b-89aa-309469cbf16f] DiffEqMonteCarlo from https://github.com/JuliaDiffEq/DiffEqMonteCarlo.jl.git
[ Info: Installed DiffEqMonteCarlo ──────────── v0.15.1
   Cloning [d96e819e-fc66-5662-9728-84c9c7592b0a] Parameters from https://github.com/mauro3/Parameters.jl.git
[ Info: Installed Parameters ────────────────── v0.10.3
   Cloning [2b5f629d-d688-5b77-993f-72d75c75574e] DiffEqBase from https://github.com/JuliaDiffEq/DiffEqBase.jl.git
[ Info: Installed DiffEqBase ────────────────── v5.12.0
   Cloning [00ebfdb7-1f24-5e51-bd34-a7502290713f] CSTParser from https://github.com/julia-vscode/CSTParser.jl.git
[ Info: Installed CSTParser ─────────────────── v0.6.0
   Cloning [d41bc354-129a-5804-8e4c-c37616107c6c] NLSolversBase from https://github.com/JuliaNLSolvers/NLSolversBase.jl.git
[ Info: Installed NLSolversBase ─────────────── v7.3.1
   Cloning [1914dd2f-81c6-5fcd-8719-6d5c9610ff09] MacroTools from https://github.com/MikeInnes/MacroTools.jl.git
[ Info: Installed MacroTools ────────────────── v0.5.0
   Cloning [46d2c3a1-f734-5fdb-9937-b9b9aeba4221] MuladdMacro from https://github.com/JuliaDiffEq/MuladdMacro.jl.git
[ Info: Installed MuladdMacro ───────────────── v0.2.1
   Cloning [0796e94c-ce3b-5d07-9a54-7f471281c624] Tokenize from https://github.com/JuliaLang/Tokenize.jl.git
[ Info: Installed Tokenize ──────────────────── v0.5.4
   Cloning [e409e4f3-bfea-5376-8464-e040bb5c01ab] PoissonRandom from https://github.com/JuliaDiffEq/PoissonRandom.jl.git
[ Info: Installed PoissonRandom ─────────────── v0.4.0
   Cloning [82899510-4779-5014-852e-03e436cf321d] IteratorInterfaceExtensions from https://github.com/queryverse/IteratorInterfaceExtensions.jl.git
[ Info: Installed IteratorInterfaceExtensions ─ v1.0.0
   Cloning [c619ae07-58cd-5f6d-b883-8f17bd6a98f9] DimensionalPlotRecipes from https://github.com/JuliaDiffEq/DimensionalPlotRecipes.jl.git
[ Info: Installed DimensionalPlotRecipes ────── v0.2.0
   Cloning [bbf7d656-a473-5ed7-a52c-81e309532950] CommonSubexpressions from https://github.com/rdeits/CommonSubexpressions.jl.git
[ Info: Installed CommonSubexpressions ──────── v0.2.0
   Cloning [a3b82374-2e81-5b9e-98ce-41277c0e4c87] MatrixFactorizations from https://github.com/JuliaMatrices/MatrixFactorizations.jl.git
[ Info: Installed MatrixFactorizations ──────── v0.0.4
   Cloning [30578b45-9adc-5946-b283-645ec420af67] URIParser from https://github.com/JuliaWeb/URIParser.jl.git
[ Info: Installed URIParser ─────────────────── v0.4.0
   Cloning [731186ca-8d62-57ce-b412-fbd966d074cd] RecursiveArrayTools from https://github.com/JuliaDiffEq/RecursiveArrayTools.jl.git
[ Info: Installed RecursiveArrayTools ───────── v0.20.0
   Cloning [e6cf234a-135c-5ec9-84dd-332b85af5143] RandomNumbers from https://github.com/sunoru/RandomNumbers.jl.git
[ Info: Installed RandomNumbers ─────────────── v1.3.0
   Cloning [f6369f11-7733-5829-9624-2563aa707210] ForwardDiff from https://github.com/JuliaDiff/ForwardDiff.jl.git
[ Info: Installed ForwardDiff ───────────────── v0.10.3
   Cloning [ae029012-a4dd-5104-9daa-d747884805df] Requires from https://github.com/MikeInnes/Requires.jl.git
[ Info: Installed Requires ──────────────────── v0.5.2
   Cloning [276daf66-3868-5448-9aa4-cd146d93841b] SpecialFunctions from https://github.com/JuliaMath/SpecialFunctions.jl.git
[ Info: Installed SpecialFunctions ──────────── v0.7.2
   Cloning [77a26b50-5914-5dd7-bc55-306e6241c503] DiffEqNoiseProcess from https://github.com/JuliaDiffEq/DiffEqNoiseProcess.jl.git
[ Info: Installed DiffEqNoiseProcess ────────── v3.3.1
   Cloning [5a0ffddc-d203-54b0-88ba-2c03c0fc2e67] DiffEqFinancial from https://github.com/JuliaDiffEq/DiffEqFinancial.jl.git
[ Info: Installed DiffEqFinancial ───────────── v2.1.0
   Cloning [189a3867-3050-52da-a836-e630ba90ab69] Reexport from https://github.com/simonster/Reexport.jl.git
[ Info: Installed Reexport ──────────────────── v0.2.0
   Cloning [f2c3362d-daeb-58d1-803e-2bc74f2840b4] RecursiveFactorization from https://github.com/YingboMa/RecursiveFactorization.jl.git
[ Info: Installed RecursiveFactorization ────── v0.0.1
   Cloning [2774e3e8-f4cf-5e23-947b-6d7e65073b56] NLsolve from https://github.com/JuliaNLSolvers/NLsolve.jl.git
[ Info: Installed NLsolve ───────────────────── v4.0.0
   Cloning [4fba245c-0d91-5ea0-9b3e-6abc04ee57a9] ArrayInterface from https://github.com/JuliaDiffEq/ArrayInterface.jl.git
[ Info: Installed ArrayInterface ────────────── v0.1.1
   Cloning [3783bdb8-4a98-5b6b-af9a-565f29a5fe9c] TableTraits from https://github.com/queryverse/TableTraits.jl.git
[ Info: Installed TableTraits ───────────────── v1.0.0
   Cloning [163ba53b-c6d8-5494-b064-1a9d43ac40c5] DiffResults from https://github.com/JuliaDiff/DiffResults.jl.git
[ Info: Installed DiffResults ───────────────── v0.0.4
   Cloning [01453d9d-ee7c-5054-8395-0335cb756afa] DiffEqDiffTools from https://github.com/JuliaDiffEq/DiffEqDiffTools.jl.git
[ Info: Installed DiffEqDiffTools ───────────── v0.13.0
   Cloning [055956cb-9e8b-5191-98cc-73ae4a59e68a] DiffEqPhysics from https://github.com/JuliaDiffEq/DiffEqPhysics.jl.git
[ Info: Installed DiffEqPhysics ─────────────── v3.1.0
   Cloning [d3d80556-e9d4-5f37-9878-2ab0fcc64255] LineSearches from https://github.com/JuliaNLSolvers/LineSearches.jl.git
[ Info: Installed LineSearches ──────────────── v7.0.1
   Cloning [789caeaf-c7a9-5a7d-9973-96adeb23e2a0] StochasticDiffEq from https://github.com/JuliaDiffEq/StochasticDiffEq.jl.git
[ Info: Installed StochasticDiffEq ──────────── v6.5.0
   Cloning [c3572dad-4567-51f8-b174-8c6c989267f4] Sundials from https://github.com/JuliaDiffEq/Sundials.jl.git
[ Info: Installed Sundials ──────────────────── v3.6.1
   Cloning [9fdde737-9c7f-55bf-ade8-46b3f136cc48] DiffEqOperators from https://github.com/JuliaDiffEq/DiffEqOperators.jl.git
[ Info: Installed DiffEqOperators ───────────── v3.5.0
   Cloning [9672c7b4-1e72-59bd-8a11-6ac3964bc41f] SteadyStateDiffEq from https://github.com/JuliaDiffEq/SteadyStateDiffEq.jl.git
[ Info: Installed SteadyStateDiffEq ─────────── v1.4.0
   Cloning [9e28174c-4ba2-5203-b857-d8d62c4213ee] BinDeps from https://github.com/JuliaPackaging/BinDeps.jl.git
[ Info: Installed BinDeps ───────────────────── v0.8.10
   Cloning [a2a6695c-b41b-5b7d-aed9-dbfdeacea5d7] TreeViews from https://github.com/pfitzseb/TreeViews.jl.git
[ Info: Installed TreeViews ─────────────────── v0.3.0
   Cloning [bcd4f6db-9728-5f36-b5f7-82caef46ccdb] DelayDiffEq from https://github.com/JuliaDiffEq/DelayDiffEq.jl.git
[ Info: Installed DelayDiffEq ───────────────── v5.4.1
   Cloning [c894b116-72e5-5b58-be3c-e6d8d4ac2b12] DiffEqJump from https://github.com/JuliaDiffEq/DiffEqJump.jl.git
[ Info: Installed DiffEqJump ────────────────── v6.1.1
  Updating `~/.julia/environments/v1.1/Project.toml`
  [0c46a032] + DifferentialEquations v6.4.0
  Updating `~/.julia/environments/v1.1/Manifest.toml`
  [4fba245c] + ArrayInterface v0.1.1
  [aae01518] + BandedMatrices v0.9.2
  [9e28174c] + BinDeps v0.8.10
  [b99e7846] + BinaryProvider v0.5.5
  [764a87c0] + BoundaryValueDiffEq v2.2.3
  [00ebfdb7] + CSTParser v0.6.0
  [49dc2e85] + Calculus v0.4.1
  [bbf7d656] + CommonSubexpressions v0.2.0
  [34da2185] + Compat v2.1.0
  [864edb3b] + DataStructures v0.15.0
  [bcd4f6db] + DelayDiffEq v5.4.1
  [2b5f629d] + DiffEqBase v5.12.0
  [459566f4] + DiffEqCallbacks v2.5.2
  [01453d9d] + DiffEqDiffTools v0.13.0
  [5a0ffddc] + DiffEqFinancial v2.1.0
  [c894b116] + DiffEqJump v6.1.1
  [78ddff82] + DiffEqMonteCarlo v0.15.1
  [77a26b50] + DiffEqNoiseProcess v3.3.1
  [9fdde737] + DiffEqOperators v3.5.0
  [055956cb] + DiffEqPhysics v3.1.0
  [163ba53b] + DiffResults v0.0.4
  [b552c78f] + DiffRules v0.0.10
  [0c46a032] + DifferentialEquations v6.4.0
  [c619ae07] + DimensionalPlotRecipes v0.2.0
  [b4f34e82] + Distances v0.8.0
  [ffbed154] + DocStringExtensions v0.8.0
  [d4d017d3] + ExponentialUtilities v1.5.1
  [1a297f60] + FillArrays v0.6.3
  [f6369f11] + ForwardDiff v0.10.3
  [069b7b12] + FunctionWrappers v1.0.0
  [01680d73] + GenericSVD v0.2.1
  [42fd0dbc] + IterativeSolvers v0.8.1
  [82899510] + IteratorInterfaceExtensions v1.0.0
  [5078a376] + LazyArrays v0.9.0
  [d3d80556] + LineSearches v7.0.1
  [1914dd2f] + MacroTools v0.5.0
  [a3b82374] + MatrixFactorizations v0.0.4
  [46d2c3a1] + MuladdMacro v0.2.1
  [f9640e96] + MultiScaleArrays v1.4.0
  [d41bc354] + NLSolversBase v7.3.1
  [2774e3e8] + NLsolve v4.0.0
  [77ba4419] + NaNMath v0.3.2
  [bac558e1] + OrderedCollections v1.1.0
  [1dea7af3] + OrdinaryDiffEq v5.8.1
  [d96e819e] + Parameters v0.10.3
  [e409e4f3] + PoissonRandom v0.4.0
  [e6cf234a] + RandomNumbers v1.3.0
  [3cdcf5f2] + RecipesBase v0.7.0
  [731186ca] + RecursiveArrayTools v0.20.0
  [f2c3362d] + RecursiveFactorization v0.0.1
  [189a3867] + Reexport v0.2.0
  [ae029012] + Requires v0.5.2
  [ae5879a3] + ResettableStacks v0.6.0
  [f2b01f46] + Roots v0.8.1
  [276daf66] + SpecialFunctions v0.7.2
  [90137ffa] + StaticArrays v0.11.0
  [9672c7b4] + SteadyStateDiffEq v1.4.0
  [789caeaf] + StochasticDiffEq v6.5.0
  [c3572dad] + Sundials v3.6.1
  [3783bdb8] + TableTraits v1.0.0
  [0796e94c] + Tokenize v0.5.4
  [a2a6695c] + TreeViews v0.3.0
  [30578b45] + URIParser v0.4.0
  [2a0f44e3] + Base64 
  [ade2ca70] + Dates 
  [8bb1440f] + DelimitedFiles 
  [8ba89e20] + Distributed 
  [b77e0a4c] + InteractiveUtils 
  [76f85450] + LibGit2 
  [8f399da3] + Libdl 
  [37e2e46d] + LinearAlgebra 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [a63ad114] + Mmap 
  [44cfe95a] + Pkg 
  [de0858da] + Printf 
  [3fa0cd96] + REPL 
  [9a3f8284] + Random 
  [ea8e919c] + SHA 
  [9e88b42a] + Serialization 
  [1a1011a3] + SharedArrays 
  [6462fe0b] + Sockets 
  [2f01184e] + SparseArrays 
  [10745b16] + Statistics 
  [4607b0f0] + SuiteSparse 
  [8dfed614] + Test 
  [cf7118a7] + UUIDs 
  [4ec0a83e] + Unicode 
  Building SpecialFunctions → `~/.julia/packages/SpecialFunctions/fvheQ/deps/build.log`
┌ Error: Error building `SpecialFunctions`: 
│ [ Info: Downloading https://github.com/JuliaMath/OpenspecfunBuilder/releases/download/v0.5.3-3/Openspecfun.v0.5.3.x86_64-linux-gnu-gcc4.tar.gz to /home/developer/.julia/packages/SpecialFunctions/fvheQ/deps/usr/downloads/Openspecfun.v0.5.3.x86_64-linux-gnu-gcc4.tar.gz...
│ ERROR: LoadError: Could not download https://github.com/JuliaMath/OpenspecfunBuilder/releases/download/v0.5.3-3/Openspecfun.v0.5.3.x86_64-linux-gnu-gcc4.tar.gz to /home/developer/.julia/packages/SpecialFunctions/fvheQ/deps/usr/downloads/Openspecfun.v0.5.3.x86_64-linux-gnu-gcc4.tar.gz:
│ ErrorException("")
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] #download#96(::Bool, ::Function, ::String, ::String) at /home/developer/.julia/packages/BinaryProvider/A0sDa/src/PlatformEngines.jl:615
│  [3] #download at ./none:0 [inlined]
│  [4] #download_verify#97(::Bool, ::Bool, ::Bool, ::Function, ::String, ::String, ::String) at /home/developer/.julia/packages/BinaryProvider/A0sDa/src/PlatformEngines.jl:684
│  [5] #download_verify at ./none:0 [inlined]
│  [6] #install#137(::Prefix, ::String, ::Bool, ::Bool, ::Bool, ::Function, ::String, ::String) at /home/developer/.julia/packages/BinaryProvider/A0sDa/src/Prefix.jl:314
│  [7] (::getfield(BinaryProvider, Symbol("#kw##install")))(::NamedTuple{(:prefix, :force, :verbose),Tuple{Prefix,Bool,Bool}}, ::typeof(install), ::String, ::String) at ./none:0
│  [8] top-level scope at /home/developer/.julia/packages/SpecialFunctions/fvheQ/deps/build.jl:142
│  [9] include at ./boot.jl:326 [inlined]
│  [10] include_relative(::Module, ::String) at ./loading.jl:1038
│  [11] include(::Module, ::String) at ./sysimg.jl:29
│  [12] include(::String) at ./client.jl:403
│  [13] top-level scope at none:0
│ in expression starting at /home/developer/.julia/packages/SpecialFunctions/fvheQ/deps/build.jl:139
│ [12:54:21] 
│ [12:54:21] curl: (7) Failed to connect to github.com port 443: Connection timed out
└ @ Pkg.Operations ~/julia-1.1.1/usr/share/julia/stdlib/v1.1/Pkg/src/Operations.jl:1075
  Building Sundials ────────→ `~/.julia/packages/Sundials/OdI66/deps/build.log`
┌ Error: Error building `Sundials`: 
│ ┌ Warning: platform_key() is deprecated, use platform_key_abi() from now on
│ │   caller = ip:0x0
│ └ @ Core :-1
│ [ Info: Downloading https://github.com/JuliaDiffEq/SundialsBuilder/releases/download/sundials-3.1.1.6/Sundials.v1.0.0.x86_64-linux-gnu.tar.gz to /home/developer/.julia/packages/Sundials/OdI66/deps/usr/downloads/Sundials.v1.0.0.x86_64-linux-gnu.tar.gz...
│ ERROR: LoadError: Could not download https://github.com/JuliaDiffEq/SundialsBuilder/releases/download/sundials-3.1.1.6/Sundials.v1.0.0.x86_64-linux-gnu.tar.gz to /home/developer/.julia/packages/Sundials/OdI66/deps/usr/downloads/Sundials.v1.0.0.x86_64-linux-gnu.tar.gz:
│ ErrorException("")
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] #download#96(::Bool, ::Function, ::String, ::String) at /home/developer/.julia/packages/BinaryProvider/A0sDa/src/PlatformEngines.jl:615
│  [3] #download at ./none:0 [inlined]
│  [4] #download_verify#97(::Bool, ::Bool, ::Bool, ::Function, ::String, ::String, ::String) at /home/developer/.julia/packages/BinaryProvider/A0sDa/src/PlatformEngines.jl:684
│  [5] #download_verify at ./none:0 [inlined]
│  [6] #install#137(::Prefix, ::String, ::Bool, ::Bool, ::Bool, ::Function, ::String, ::String) at /home/developer/.julia/packages/BinaryProvider/A0sDa/src/Prefix.jl:314
│  [7] (::getfield(BinaryProvider, Symbol("#kw##install")))(::NamedTuple{(:prefix, :force, :verbose),Tuple{Prefix,Bool,Bool}}, ::typeof(install), ::String, ::String) at ./none:0
│  [8] top-level scope at /home/developer/.julia/packages/Sundials/OdI66/deps/build.jl:53
│  [9] include at ./boot.jl:326 [inlined]
│  [10] include_relative(::Module, ::String) at ./loading.jl:1038
│  [11] include(::Module, ::String) at ./sysimg.jl:29
│  [12] include(::String) at ./client.jl:403
│  [13] top-level scope at none:0
│ in expression starting at /home/developer/.julia/packages/Sundials/OdI66/deps/build.jl:49
│ [12:56:36] 
│ [12:56:36] curl: (7) Failed to connect to github.com port 443: Connection timed out
└ @ Pkg.Operations ~/julia-1.1.1/usr/share/julia/stdlib/v1.1/Pkg/src/Operations.jl:1075
```
I must say that I am actually behind a corporate proxy, I have the following _git_ configuration:

>http.proxy=… (my proxy details)
>url.https://.insteadof=git://
>https.proxy=… (my proxy details)

I built Julia 1.1.1 from the .tar.gz file available at https://julialang.org/downloads/ at "Tarball with dependencies".

Prior to the compilation,  I installed the following general packages:

>sudo apt-get install build-essential libatomic1 python gfortran perl wget m4 cmake pkg-config
>sudo apt-get install original-awk libbsd-dev manpages-dev

The following libgit-related packages:

>sudo apt-get install libgit2-dev

Curl-related packages:

>sudo apt-get install curl libcurl3 libcurl3-dbg libcurl3-gnutls libcurl3-nss libcurl-ocaml
>sudo apt-get install libcurl4-doc libcurl4-openssl-dev

SSL-related packages:

>sudo apt-get install openssl libssl-dev libssl-doc libssl-ocaml libssl-ocaml-dev libssl1.0.2 libssl1.1

How can I properly install DifferentialEquations? Am I doing anything wrong, or missing something?

**Comment by StefanKarpinski** [comment]
@staticfloat, looks like a BB download problem. Any ideas?

**Comment by staticfloat** [comment]
```
[12:54:21] curl: (7) Failed to connect to github.com port 443: Connection timed out
```

Looks like a network hiccup to me.

**Comment by rgalarcia** [comment]
I don't think the cause of the error is merely a network hiccup. It happened all the times I tested yesterday (3 or 4 times over the course of 2 hours), and it happened again today. In fact, I could access _github_'s package URL with a browser at all times and I could manually download the .tar.gz files without any problem.

I solved the issue manually with the following steps:

- I manually downloaded, using a browser, the file https://github.com/JuliaMath/OpenspecfunBuilder/releases/download/v0.5.3-3/Openspecfun.v0.5.3.x86_64-linux-gnu-gcc4.tar.gz and I moved it to /home/developer/.julia/packages/SpecialFunctions/fvheQ/deps/usr/downloads/

- I did the same for https://github.com/JuliaDiffEq/SundialsBuilder/releases/download/sundials-3.1.1.6/Sundials.v1.0.0.x86_64-linux-gnu.tar.gz moving it to /home/developer/.julia/packages/Sundials/OdI66/deps/usr/downloads/

- I used the command _Pkg.build("SpecialFunctions")_ and then _Pkg.build("Sundials")_. As the package files were already downloaded in the required folder, Pkg did not attempt to download them again, and just built the packages.

- Finally, I used the command _using DifferentialEquations_ and the package precompiled properly, apparently (some Warnings appeared but I think they are not important).

I tested the examples listed in http://docs.juliadiffeq.org/latest/tutorials/ode_example.html and it seems they worked well. I had to do the same manual download procedure to install the package Plots, however, as it depends on the package GR, which also failed to download automatically.

**Comment by KristofferC** [comment]
The

```
   Cloning [01453d9d-ee7c-5054-8395-0335cb756afa] DiffEqDiffTools from https://github.com/JuliaDiffEq/DiffEqDiffTools.jl.git
```

stuff in the installation log also shows that it fails to download the tarballs and falls back to using full git clones instead.

**Comment by staticfloat** [comment]
Perhaps there is something broken with your `curl`, or whatever download engine it is using?  From the command line, can you use `curl -v -L <url> -O <filepath>` to download things?

**Comment by rgalarcia** [comment]
I finally found the error. The `curl -v -L <url> -O <filepath>` did not work at first, as I expected. The problem was that I had forgotten to add the global proxy configuration to the machine (in Debian 9.9, accessible via System -> Preferences -> Internet and Network ->Network Proxy). 

The proxies were only configured for particular things such as _github_ (through `git config --global http.proxy` and `git config --global https.proxy` commands) and _Firefox_ (inside the app, via Preferences -> General -> Network Proxy). 

This is why the git clones were working, and this is why I could manually download the .tar.gz files with the browser without any problem, but the curl failed to download them. After adding the required global configuration to the machine, the `curl -v -L <url> -O <filepath>` test worked, and the issue was solved. Thank you very much for your help.
